### PR TITLE
Delegate to LoadStratgy for module information

### PR DESCRIPTION
**Pull request by grouma**
- LoadStrategy now has the following:
  - `moduleForServerPath`
  - `serverPathForModule`
  - `serverPathForAppUri`
- Since there is a flavor of `LoadStrategy` for each compiler, we now have consistency with handling module names and no longer leak Google3 details externally (we'll have a Google3 specific LegacyStrategy internaly)
- Fix Frontend Server tests locally, in a follow up PR I'll enable them on Travis for the dev SDK
- Fix Readme tests due to `package:arg` update

Towards https://github.com/dart-lang/webdev/issues/910
Closes https://github.com/dart-lang/webdev/issues/917

### CODE REVIEW

**Comment by grouma** [comment]
cc @jonahwilliams

This change along with a minor fix to Flutter Tools will ensure `org-dartlang-app` entries are loaded. In `devfs_web.dart` you need to look for cached files here:
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_runner/devfs_web.dart#L385

The serverPath will now be the fully qualified path assuming you leverage this (or do something similar):
https://github.com/dart-lang/webdev/pull/925/files#diff-9e9614b8283ba14570d802d5eaa4b132R53

**Comment by jonahwilliams** [comment]
Nice!

File: dwds/lib/src/loaders/build_runner_require.dart [hunk 1]
```diff
@@ -50,8 +60,32 @@ class BuildRunnerRequireStrategyProvider {
...
+    }
+    return result;
+  }
+
+  String _moduleForServerPath(String serverPath) {
+    if (!serverPath.endsWith('$_extension.js')) return null;
+    serverPath =
+        serverPath.startsWith('/') ? serverPath.substring(1) : serverPath;
+    // Remove the .js from the path.
+    serverPath = p.withoutExtension(serverPath);
```
Comment (jakemac53) [inline dwds/lib/src/loaders/build_runner_require.dart]: a bit surprising we don't remove the full extension here?

Comment (grouma) [reply dwds/lib/src/loaders/build_runner_require.dart]: Our server path is of the form `packages/path/path.ddc`. It's commented above so it shouldn't be surprising. Should I add more comments?

Comment (jakemac53) [reply dwds/lib/src/loaders/build_runner_require.dart]: The comment is just repeating the code and not explaining why it is the way it is :D. A follow-up to clarify would be good but not required.

Comment (grouma) [reply dwds/lib/src/loaders/build_runner_require.dart]: I meant here: https://github.com/dart-lang/webdev/blob/master/dwds/lib/src/loaders/build_runner_require.dart#L52

**Pull request merged**